### PR TITLE
ACT labs map etc.

### PR DESCRIPTION
**Pull request by dckc**
this has conflicts? I wonder what they are...

### CODE REVIEW

**Review by nihr43 — APPROVED** [review]

File: act/Makefile [hunk 1]
```diff
@@ -3,39 +3,58 @@
...
+status/visit_details:
+	sql $(ACCOUNT) @visit_details_map.sql \
+		$(metadata_schema) $(shrine_ont_schema) $(heron_data_schema)
 	mkdir -p status
-	echo "These files represent completion status of SQL scripts." >$@
+	touch $@
 
 status/demographics: status/meta_manual_mapping
-	sqlplus $(ACCOUNT) @demographics_mapping.sql $(metadata_schema) $(shrine_ont_schema)
+	sql $(ACCOUNT) @demographics_mapping.sql $(metadata_schema) $(shrine_ont_schema)
```
Comment (lav-patel) [inline act/Makefile:24]: META_MANUAL_MAPPING.csv is being used in demographics_mapping.sql, so I would request to do that step first.

File: act/onto_index.sql [hunk 2]
```diff
@@ -7,14 +7,10 @@ drop index SHRINE_ONT_ACT.ACT_COVID_fullname_IDX;
 drop index SHRINE_ONT_ACT.ACT_COVID_hlevel_IDX;
 drop index SHRINE_ONT_ACT.ACT_DEMO_APPLIED_IDX ;
```
Comment (lav-patel) [inline act/onto_index.sql]: We should move all of these indexes to appropriate sql mapping to maintain the new format.

File: act/update_concept_dimension.sql [hunk 3]
```diff
@@ -71,7 +71,7 @@ DELETE /*+  PARALLEL (20) */ from
 "&&metadata_schema"."HERON_TERMS"
 where C_FULLNAME = '\i2b2\Demographics\LESS_THAN_10\'
```
Comment (lav-patel) [inline act/update_concept_dimension.sql]: Can you get rid of the files as you are building concpet_dimension in each SQL file? ?

File: act/create_metadata_tables_from_shrine_ont.sql [hunk 4]
```diff
@@ -14,7 +14,7 @@ drop table "&&metadata_schema".ACT_LOINC_LAB_2018AA purge;
 drop table "&&metadata_schema".ACT_MED_ALPHA_V2_121318 purge;
 drop table "&&metadata_schema".ACT_MED_VA_V2_092818 purge;
 drop table "&&metadata_schema".NCATS_LABS purge;
```
Comment (lav-patel) [inline act/create_metadata_tables_from_shrine_ont.sql]: move labs statment its own lab mapping file

File: act/create_metadata_tables_from_shrine_ont.sql [hunk 5]
```diff
@@ -24,6 +24,6 @@ create table "&&metadata_schema".ACT_LOINC_LAB_2018AA  nologging as select * fro
 create table "&&metadata_schema".ACT_MED_ALPHA_V2_121318  nologging as select * from "&&shrine_ont_schema".ACT_MED_ALPHA_V2_121318 ;
 create table "&&metadata_schema".ACT_MED_VA_V2_092818  nologging as select * from "&&shrine_ont_schema".ACT_MED_VA_V2_092818 ;
 create table "&&metadata_schema".NCATS_LABS  nologging as select * from "&&shrine_ont_schema".NCATS_LABS ;
```
Comment (lav-patel) [inline act/create_metadata_tables_from_shrine_ont.sql]: same as above comment

File: act/Makefile [hunk 6]
```diff
@@ -3,39 +3,58 @@
...
+		$(metadata_schema) $(shrine_ont_schema) $(etl_schema) ACT_MED_VA_V2_092818
+	mkdir -p status
 	touch $@
 
-status/table_access: status/demographics status/diagnosis status/procedures
-	sqlplus $(ACCOUNT) @update_table_access.sql $(metadata_schema)
+status/procedures:
+	sql $(ACCOUNT) @procedure_ont_map.sql $(metadata_schema)
+	mkdir -p status
 	touch $@
```
Comment (lav-patel) [inline act/Makefile:58]: Dont' see  building COVID metadata using makefile. is that intentional ?

**Review by lav-patel — CHANGES_REQUESTED** [review]

File: act/onto_index.sql (hunk 2 above)
Comment (dckc) [reply act/onto_index.sql]: yes... I guess I didn't quite finish that.

File: act/update_concept_dimension.sql (hunk 3 above)
Comment (dckc) [reply act/update_concept_dimension.sql]: yes; I didn't quite finish that either, did I?

File: act/Makefile (hunk 6 above)
Comment (dckc) [reply act/Makefile:58]: no. oops.

File: act/Makefile (hunk 1 above)
Comment (dckc) [reply act/Makefile:24]: That's what line 22 says, right? `status/demographics` depends on `status/meta_manual_mapping`. `make` ensures that dependencies are satisfied before this stanza is run.

File: act/Makefile (hunk 6 above)
Comment (dckc) [reply act/Makefile:58]: fixed in ba3ba63

**Comment by dckc** [comment]
Lav, does it look right now?

File: act/demographics_mapping.sql [hunk 7]
```diff
@@ -151,3 +151,22 @@ insert /*+ APPEND */ into nightherondata.concept_dimension (
 commit;
```
Comment (lav-patel) [inline act/demographics_mapping.sql:152]: --- land act_covid concepts in concept_dimension ; missleading comment at line 126

**Review by lav-patel — APPROVED** [review]
looks good.

**Pull request merged**